src/main/java/fi/helsinki/cs/tmc/core/communication/TmcServerCommunicationTaskFactory.java

```diff
@@ -78,7 +81,33 @@ public void setSettings(TmcSettings settings) {
         this.settings = settings;
     }
 
-    private URI getCourseListUrl() {
+    /**
+     * Returns a Callable that calls the given Callable.
+     *
+     * <p>If the call fails, the auth is refreshed and the call is done again.</p>
+     *
+     * @param <T> return type of the callable
+     * @param callable Callable to be wrapped
+     * @return Callable that wraps the given callable
+     */
+    private <T> Callable<T> tryCallCallable(final Callable<T> callable) {
+        if (!settings.getUsingOauth()) {
+            return callable;
+        }
+        return new Callable<T>() {
+            @Override
+            public T call() throws Exception {
+                try {
+                    return callable.call();
+                } catch (Exception e) {
+                    oauth.refreshToken();
```

Comment:
Can you be more specific about what you mean with logging?
Do you mean using this? https://github.com/testmycode/tmc-core/blob/c54d76b6f0430404b1d2e564f3754a24fc3746a6/src/main/java/fi/helsinki/cs/tmc/spyware/EventStore.java#L26 or what with should it be logged?